File: app/src/main/java/ca/ivasenko/mobiletest/categories/data/CategoryDao.java
package ca.ivasenko.mobiletest.categories.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import ca.ivasenko.mobiletest.categories.model.Category;

@Dao
public interface CategoryDao
{
    @Query("SELECT * from categories")
    LiveData<List<Category>> getAllCategories();

    @Query("SELECT * from categories WHERE id = :categoryId")
    LiveData<Category> getCategoryById(int categoryId);

    @Insert
    long insert(Category category);

    @Insert
    void insertList(List<Category> list);

    @Delete
    void deleteCategory(Category category);
}

File: app/src/main/java/com/rodionov/igor/mobiletest/data/CategoryDao.kt
package com.rodionov.igor.mobiletest.data

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*

import com.rodionov.igor.mobiletest.model.Category

@Dao
interface CategoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    @Transaction
    fun insert(categoryList: List<Category>)

    @Query("SELECT * FROM categories")
    fun fetchAll(): LiveData<List<Category>>
}
